firmware/src/widgets/stockWidget.cpp

```diff
@@ -108,29 +112,66 @@ void StockWidget::displayStock(int8_t displayIndex, StockDataModel &stock, uint3
     TFT_eSPI &display = m_manager.getDisplay();
 
     display.fillScreen(TFT_BLACK);
-    display.setTextColor(TFT_WHITE);
-    display.setTextSize(4);  // Set text size
+
 
     // Calculate center positions
     int screenWidth = display.width();
     int centre = 120;
-
-    // Draw stock data
-    display.fillRect(0, 0, screenWidth, 50, 0x0256);  // rgb565 colors
-    display.drawString(stock.getSymbol(), centre, 27, 1);
-    display.drawString("$" + stock.getCurrentPrice(2), centre, 51 + display.fontHeight(1), 1);
-
+    int arrowOffsetX = 0;
+    int arrowOffsetY = -109;
+
+    // ensuring proper currency display
+    String currencySymbol = "$";
+    String currencySymbolEUR = "EUR";
+    String currencySymbolGBP = "GBP";
+    // the below are just placeholders untill we have a unicode font that supports the currency symbols, which is why we're not just using the direct value from getCurrency.
+    // additionally, in some cases(crypto) the ticker wont return a currency value, so it must be extracted from the symbol deffinition 
+    if (stock.getCurrency() == "EUR"){
+       currencySymbol = currencySymbolEUR;
+    } else if (stock.getCurrency() == "GBP"){
+       currencySymbol = currencySymbolGBP;
+    } else if (stock.getSymbol().indexOf("/EUR") != -1){
+       currencySymbol = currencySymbolEUR;
+    } else if (stock.getSymbol().indexOf("/GBP") != -1){
+       currencySymbol = currencySymbolGBP;
+    }
```

Comment:
Wouldn't it be simpler to do: `currencySymbol = stock.getCurrency()` for now?
Then it works the same for all currencies, and it isn't only 'fixed' for EUR and GBP.

Reply:
Yes, but I will be fixing this very soon here, so I will be leaving it for now.

Reply:
All of this logic should be in the data model.